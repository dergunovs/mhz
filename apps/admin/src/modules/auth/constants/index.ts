export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
