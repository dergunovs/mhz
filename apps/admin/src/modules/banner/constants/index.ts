export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
