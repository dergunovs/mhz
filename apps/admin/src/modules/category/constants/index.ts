export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
