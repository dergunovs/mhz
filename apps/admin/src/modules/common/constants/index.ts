export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
