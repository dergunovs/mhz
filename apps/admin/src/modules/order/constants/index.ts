export const URL_ORDER = '/orders';
