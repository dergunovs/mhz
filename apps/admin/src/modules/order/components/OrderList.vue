<template>
  <UiTable
    :headers="tableHeaders"
    :isLoading="!props.orders?.length"
    :modelValue="props.modelValue"
    @update:modelValue="(value) => emit('update:modelValue', value)"
    @reset="(value) => emit('reset', value)"
  >
    <template v-if="props.orders?.length">
      <tr v-for="order in props.orders" :key="order._id">
        <td data-grow>
          <RouterLink :to="`${URL_ORDER}/${order._id}`">
            {{ order._id }}
          </RouterLink>
        </td>
        <td data-no-wrap>{{ order.customer.firstName }} {{ order.customer.lastName }}</td>
        <td data-no-wrap>{{ order.status }}</td>
        <td data-no-wrap>
          {{ formatDate(order.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(order.dateUpdated) }}
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { IOrder } from 'mhz-types';
import { UiTable } from 'mhz-ui';
import { formatDate, ISortOption } from 'mhz-helpers';

import { URL_ORDER } from '@/order/constants';

interface IProps {
  orders?: IOrder[];
  modelValue: ISortOption;
}

const props = defineProps<IProps>();
const emit = defineEmits(['update:modelValue', 'reset']);

const tableHeaders = [
  { value: '_id', title: 'ID' },
  { value: 'customer', title: 'Customer' },
  { value: 'status', title: 'Status' },
  { value: 'dateCreated', title: 'Created' },
  { value: 'dateUpdated', title: 'Updated' },
];
</script>
