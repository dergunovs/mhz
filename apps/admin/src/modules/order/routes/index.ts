import { URL_ORDER } from '@/order/constants';

export const orderRoutes = [
  {
    path: URL_ORDER,
    name: 'OrderList',
    component: () => import('@/order/pages/OrderListPage.vue'),
  },
  {
    path: `${URL_ORDER}/:order`,
    name: 'Order',
    component: () => import('@/order/pages/OrderPage.vue'),
  },
];
