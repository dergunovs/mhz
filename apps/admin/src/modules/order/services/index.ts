import { ComputedRef, Ref } from 'vue';

import { IOrder, TOrderStatus } from 'mhz-types';
import { api, useQuery, useMutation, IPageQuery } from 'mhz-helpers';

import { API_ORDER } from '@/order/constants';

export function getOrders(query: Ref<IPageQuery | number>) {
  async function fn(): Promise<{ data: IOrder[]; total: number }> {
    const params =
      typeof query.value === 'number'
        ? { page: query.value }
        : {
            page: query.value.page || 1,
            sort: query.value.sort.value,
            dir: query.value.sort.isAsc === false ? 'desc' : 'asc',
          };

    const { data } = await api.get(API_ORDER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_ORDER, query], queryFn: fn });
}

export function getOrder(id?: ComputedRef<string | string[]>) {
  async function fn(): Promise<IOrder | null> {
    if (!id?.value) return null;

    const { data } = await api.get(`${API_ORDER}/${id.value}`);

    return data;
  }

  return useQuery({ queryKey: [API_ORDER, id], queryFn: fn });
}

export function updateOrder(id: ComputedRef<string | undefined>, options: object) {
  async function fn(status: TOrderStatus) {
    await api.patch(`${API_ORDER}/${id.value}`, { status });
  }

  return useMutation({ mutationKey: [API_ORDER, id], mutationFn: fn, ...options });
}

export function deleteOrder(id: ComputedRef<string | undefined>, options: object) {
  async function fn() {
    await api.delete(`${API_ORDER}/${id.value}`);
  }

  return useMutation({ mutationKey: [API_ORDER], mutationFn: fn, ...options });
}
