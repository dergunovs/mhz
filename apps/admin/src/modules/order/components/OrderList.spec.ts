import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import OrderList from './OrderList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_ORDER } from '@/order/constants';
import { ORDERS } from '@/order/fixtures';

const orderListRow = dataTest('order-list-row');
const orderListLink = dataTest('order-list-link');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(OrderList, {
    props: {
      orders: ORDERS.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('OrderList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(OrderList)).toBeTruthy();
  });

  it('shows rows if orders props', async () => {
    expect(wrapper.findAll(orderListRow).length).toEqual(ORDERS.data.length);

    await wrapper.setProps({ orders: [] });

    expect(wrapper.find(orderListRow).exists()).toBe(false);
  });

  it('generated link to order', async () => {
    const LINK = `${URL_ORDER}/${ORDERS.data[0]._id}`;

    expect(wrapper.find(orderListLink).attributes('to')).toEqual(LINK);
  });

  it('shows order content in cells', async () => {
    expect(wrapper.find(orderListLink).text()).toEqual(ORDERS.data[0]._id);
  });
});
