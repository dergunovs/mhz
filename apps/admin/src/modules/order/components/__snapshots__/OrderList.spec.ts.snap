// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" isloading="false" lang="en">
  <tbody>
    <tr data-test="order-list-row">
      <td data-grow=""><a to="/orders/65940c82cc2989495f96d911" data-test="order-list-link">65940c82cc2989495f96d911</a></td>
      <td data-no-wrap="">352000 ₽</td>
      <td data-no-wrap="">Alexandr Dergunov</td>
      <td>new</td>
      <td data-no-wrap="">Jan 2, 2024</td>
      <td data-no-wrap="">Jan 2, 2024</td>
    </tr>
    <tr data-test="order-list-row">
      <td data-grow=""><a to="/orders/6587ec081614096a5773abdb" data-test="order-list-link">6587ec081614096a5773abdb</a></td>
      <td data-no-wrap="">29900 ₽</td>
      <td data-no-wrap="">Alexandr Dergunov</td>
      <td>completed</td>
      <td data-no-wrap="">Dec 24, 2023</td>
      <td data-no-wrap="">Dec 30, 2023</td>
    </tr>
  </tbody>
</table>"
`;
